### PR TITLE
buttons

### PR DESCRIPTION
**Pull request by ChrisEddy**

### CODE REVIEW

File: planet.js [hunk 1]
```diff
@@ -777,23 +777,34 @@ planet.connect = async (rn, {online = true} = {}) => {
...
         channelConnectionOpen = false;
+        presenceHost = '';
       } else {
-        planet.connect('lol');
-        button.innerHTML = `
-          <i class="fal fa-wifi-slash"></i>
-          <div class=label>Disconnect</div>
-        `;
-      }
+          const response = fetch('https://worlds.exokit.org/create', {
```
Comment (avaer) [inline planet.js:795]: Probably should be a constant.

File: planet.js [hunk 2]
```diff
@@ -777,23 +777,34 @@ planet.connect = async (rn, {online = true} = {}) => {
 window.addEventListener('load', () => {
   const button = document.getElementById('connectButton');
   if (button) {
-    document.getElementById('connectButton').addEventListener('click', (e) => {
+    document.getElementById('connectButton').addEventListener('click', async (e) => {
```
Comment (avaer) [inline planet.js:780]: This could probably be moved outside of window load.

File: planet.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ import {makePromise} from './util.js';
 // import * as THREE from './three.module.js';
 // import { makeTextMesh } from './vr-ui.js';
 
-const presenceHost = `wss://${document.location.hostname}:4443`;
+let presenceHost = '';
```
Comment (avaer) [inline planet.js:18]: Does this even need to be a variable up here anymore?

File: planet.js (hunk 1 above)
Comment (ChrisEddy) [reply planet.js]: fixed

File: planet.js (hunk 3 above)
Comment (ChrisEddy) [reply planet.js]: fixed

File: planet.js (hunk 2 above)
Comment (ChrisEddy) [reply planet.js:782]: probs fine

**Comment by ChrisEddy** [comment]
this works now

File: planet.js (hunk 2 above)
Comment (avaer) [reply planet.js:782]: No it's not.

Comment (avaer) [reply planet.js:782]: Is there a reason for this timing?

File: planet.js [hunk 4]
```diff
@@ -10,13 +10,11 @@ import {
 } from './constants.js';
 import {XRChannelConnection} from 'https://2.metartc.com/xrrtc.js';
 import {loginManager} from './login.js';
-import {storageHost} from './constants.js';
+import { storageHost, worldsHost} from './constants.js';
```
Comment (avaer) [inline planet.js:13]: Spacing.

File: edit.js [hunk 5]
```diff
@@ -7384,8 +7393,10 @@ window.addEventListener('keydown', e => {
       break;
     }
     case 67: { // C
-      document.querySelector('.weapon[weapon="build"]').click();
-      buildMode = 'stair';
+      if (!keys.ctrl && document.pointerLockElement) {
```
Comment (avaer) [inline edit.js:7396]: This breaks camera controls.

File: edit.js [hunk 6]
```diff
@@ -7233,6 +7233,7 @@ const keys = {
   left: false,
   right: false,
   shift: false,
+  ctrl: false
```
Comment (avaer) [inline edit.js:7236]: There is probably no need to track the ctrl key.

**Review by avaer — CHANGES_REQUESTED** [review]
I think this breaks the controls, in addition to having binding races.

File: planet.js (hunk 4 above)
Comment (ChrisEddy) [reply planet.js]: fixed

**Review by avaer — APPROVED** [review]

**Pull request merged**